src/Tracing/GuzzleTracingMiddleware.php

```diff
@@ -13,27 +17,58 @@
  */
 final class GuzzleTracingMiddleware
 {
-    public static function trace(?HubInterface $hub = null): \Closure
+    public static function trace(?HubInterface $hub = null): Closure
     {
-        return function (callable $handler) use ($hub): \Closure {
-            return function (RequestInterface $request, array $options) use ($hub, $handler) {
+        return static function (callable $handler) use ($hub): Closure {
+            return static function (RequestInterface $request, array $options) use ($hub, $handler) {
                 $hub = $hub ?? SentrySdk::getCurrentHub();
                 $span = $hub->getSpan();
-                $childSpan = null;
 
-                if (null !== $span) {
-                    $spanContext = new SpanContext();
-                    $spanContext->setOp('http.guzzle');
-                    $spanContext->setDescription($request->getMethod() . ' ' . $request->getUri());
-
-                    $childSpan = $span->startChild($spanContext);
+                if (null === $span) {
+                    return $handler($request, $options);
                 }
 
-                $handlerPromiseCallback = static function ($responseOrException) use ($childSpan) {
-                    if (null !== $childSpan) {
-                        $childSpan->finish();
+                $spanContext = new SpanContext();
+                $spanContext->setOp('http.client');
+                $spanContext->setDescription($request->getMethod() . ' ' . $request->getUri());
+
+                $childSpan = $span->startChild($spanContext);
+
+                $request->withHeader('sentry-trace', $childSpan->toTraceparent());
+
+                $handlerPromiseCallback = static function ($responseOrException) use ($hub, $request, $childSpan) {
+                    $childSpan->finish();
+
+                    $response = $responseOrException instanceof ResponseInterface ? $responseOrException : null;
+
+                    /** @psalm-suppress UndefinedClass */
```

Comment:
baseline is for phpstan, this is a psalm suppression.

See also for example: https://github.com/getsentry/sentry-php/blob/bfab90c7ef921221b52b75b5108201e87f0f4661/src/HttpClient/HttpClientFactory.php#L125-L139.

Reply:
I didn't pay enough attention to the prefix and I though it was `phpstan-`. Psalm has the baseline feature too, we just never used it because I think that at the time I didn't knew it existed or because it wasn't available yet. Anyway, let's leave it as-is for now: in the future we may open a PR to generate the baseline and remove all the suppression comments at once